pkg/server/plugin/keymanager/azurekms/client_fake.go

```diff
@@ -0,0 +1,435 @@
+	KeyBundle  azkeys.KeyBundle
+	PrivateKey crypto.Signer
+}
+
+func newKMSClientFake(t *testing.T, vaultURI string, trustDomain string, serverID string, c *clock.Mock) *kmsClientFake {
+	return &kmsClientFake{
+		t:               t,
+		vaultURI:        vaultURI,
+		trustDomain:     trustDomain,
+		serverID:        serverID,
+		store:           newFakeStore(c),
+		fakeKeyIDPrefix: vaultURI + "keys/spire-key-" + serverID + "-",
+	}
+}
+
+func newFakeStore(c *clock.Mock) fakeStore {
+	return fakeStore{
+		fakeKeys: make(map[string]*fakeKeyEntry),
+		clk:      c,
+	}
+}
+
+func (fs *fakeStore) SaveKeyEntry(input *fakeKeyEntry) {
+	fs.mu.Lock()
+	defer fs.mu.Unlock()
+
+	fs.fakeKeys[input.KeyBundle.Key.KID.Name()] = input
+}
+
+func (fs *fakeStore) DeleteKeyEntry(keyName string) {
+	fs.mu.Lock()
+	defer fs.mu.Unlock()
+	delete(fs.fakeKeys, keyName)
+}
+
+func (k *kmsClientFake) setEntries(entries []fakeKeyEntry) {
+	k.mu.Lock()
+	defer k.mu.Unlock()
+	if entries == nil {
+		return
+	}
+	for _, e := range entries {
+		if e.KeyBundle.Key != nil && e.KeyBundle.Key.KID != nil && e.KeyBundle.Key.KID.Name() != "" {
+			newEntry := e
+			k.store.SaveKeyEntry(&newEntry)
+		}
+	}
+}
+
+func (k *kmsClientFake) setCreateKeyErr(fakeError string) {
+	k.mu.Lock()
+	defer k.mu.Unlock()
+	if fakeError != "" {
+		k.createKeyErr = errors.New(fakeError)
+	}
+}
+func (k *kmsClientFake) setGetKeyErr(fakeError string) {
+	k.mu.Lock()
+	defer k.mu.Unlock()
+	if fakeError != "" {
+		k.getKeyErr = errors.New(fakeError)
+	}
+}
+
+func (k *kmsClientFake) setGetPublicKeyErr(fakeError string) {
+	k.mu.Lock()
+	defer k.mu.Unlock()
+	if fakeError != "" {
+		k.getPublicKeyErr = errors.New(fakeError)
+	}
+}
+
+func (k *kmsClientFake) setUpdateKeyErr(fakeError string) {
+	k.mu.Lock()
+	defer k.mu.Unlock()
+	if fakeError != "" {
+		k.updateKeyErr = errors.New(fakeError)
+	}
+}
+
+func (k *kmsClientFake) setDeleteKeyErr(fakeError error) {
+	k.mu.Lock()
+	defer k.mu.Unlock()
+	if fakeError != nil {
+		k.deleteKeyErr = fakeError
+	}
+}
+
+func (k *kmsClientFake) setSignDataErr(fakeError string) {
+	k.mu.Lock()
+	defer k.mu.Unlock()
+	if fakeError != "" {
+		k.signErr = errors.New(fakeError)
+	}
+}
+
+func (k *kmsClientFake) setListKeysErr(fakeError string) {
+	k.mu.Lock()
+	defer k.mu.Unlock()
+	if fakeError != "" {
+		k.listKeysErr = errors.New(fakeError)
+	}
+}
+
+func (k *kmsClientFake) CreateKey(ctx context.Context, keyName string, parameters azkeys.CreateKeyParameters, options *azkeys.CreateKeyOptions) (azkeys.CreateKeyResponse, error) {
+	k.mu.RLock()
+	defer k.mu.RUnlock()
+	if k.createKeyErr != nil {
+		return azkeys.CreateKeyResponse{}, k.createKeyErr
+	}
+
+	var publicKey *azkeys.JSONWebKey
+	var privateKey crypto.Signer
+	keyOperations := getKeyOperations()
+	kmsKeyID := path.Join(k.vaultURI, keyName)
+	switch {
+	case *parameters.Kty == azkeys.JSONWebKeyTypeEC && *parameters.Curve == azkeys.JSONWebKeyCurveNameP256:
+		privateKey = k.testKeys.NewEC256(k.t)
```

Comment:
with this approach, every time we call `testKeys.NewXXXXX` we are asking for a new key to fixture (that is why you had to add more entries to fixture) so every time we want to add a new test case we'll need to add new keys.

It looks like an overkill we dont want to test key generation but that a key is returned, so what do you think about loading keys when fake is created and just use the same keys (depending on key types of course)

Reply:
Done 👍🏾